### PR TITLE
driver: watchdog: Error handling issues in wdt_wwdg_stm32.c

### PR DESCRIPTION
**Pull request by FRASTM**
This patch changes wwdg_stm32_get_pclk function to test the return value of clock_control_get_rate
It fixes the issue Error handling issues in /drivers/watchdog/wdt_wwdg_stm32.c #20503 
[Coverity CID :205655]

Signed-off-by: Francois Ramu <francois.ramu@st.com>

### CODE REVIEW

**Review by erwango — CHANGES_REQUESTED** [review]
I'm good with the change, but let's extend this to other drivers too.

**Comment by FRASTM** [comment]
Completes the fix of Error handling issues in /drivers/watchdog/wdt_wwdg_stm32.c [#20503]
applying the test of retrun code clock_control_get_rate in other drivers:  i2c, pwm, uart, spi

**Comment by FRASTM** [comment]
include logging/log.h

File: drivers/watchdog/wdt_wwdg_stm32.c [hunk 1]
```diff
@@ -84,7 +90,7 @@ static u32_t wwdg_stm32_get_timeout(struct device *dev, u32_t prescaler,
 				    u32_t counter)
 {
 	u32_t divider = WWDG_INTERNAL_DIVIDER * (1 << (prescaler >> 7));
-	float f_wwdg = (float)wwdg_stm32_get_pclk(dev) / divider;
+	float f_wwdg = wwdg_stm32_get_pclk(dev) / divider;
```
Comment (erwango) [inline drivers/watchdog/wdt_wwdg_stm32.c]: This change is not part of this issue.

File: drivers/spi/spi_ll_stm32.c [hunk 2]
```diff
@@ -281,9 +284,11 @@ static int spi_stm32_configure(struct device *dev,
...
 	}
 
-	clock_control_get_rate(device_get_binding(STM32_CLOCK_CONTROL_NAME),
-			       (clock_control_subsys_t) &cfg->pclken, &clock);
-
+	if (clock_control_get_rate(device_get_binding(STM32_CLOCK_CONTROL_NAME),
+			(clock_control_subsys_t) &cfg->pclken, &clock) < 0) {
+		LOG_ERR("Failed call clock_control_get_rate");
+		return -EIO;
+	}
```
Comment (erwango) [inline drivers/spi/spi_ll_stm32.c]: Break a line after `}`

File: drivers/pwm/pwm_stm32.c [hunk 3]
```diff
@@ -160,9 +163,11 @@ static int pwm_stm32_get_cycles_per_sec(struct device *dev, u32_t pwm,
...
 
 	/* Timer clock depends on APB prescaler */
-	clock_control_get_rate(data->clock,
-			(clock_control_subsys_t *)&cfg->pclken, &bus_clk);
-
+	if (clock_control_get_rate(data->clock,
+			(clock_control_subsys_t *)&cfg->pclken, &bus_clk) < 0) {
+		LOG_ERR("Failed call clock_control_get_rate");
+		return -EIO;
+	}
```
Comment (erwango) [inline drivers/pwm/pwm_stm32.c]: Break a line after `}`

File: drivers/i2c/i2c_ll_stm32.c [hunk 4]
```diff
@@ -14,7 +14,6 @@
 #include <drivers/i2c.h>
 #include "i2c_ll_stm32.h"
 
-#define LOG_LEVEL CONFIG_I2C_LOG_LEVEL
```
Comment (erwango) [inline drivers/i2c/i2c_ll_stm32.c]: This would deserve a dedicated clean up commit.

**Review by erwango — CHANGES_REQUESTED** [review]

File: drivers/spi/spi_ll_stm32.c (hunk 2 above)
Comment (FRASTM) [reply drivers/spi/spi_ll_stm32.c]: ok, done

File: drivers/pwm/pwm_stm32.c (hunk 3 above)
Comment (FRASTM) [reply drivers/pwm/pwm_stm32.c]: ok, done

File: drivers/i2c/i2c_ll_stm32.c (hunk 4 above)
Comment (FRASTM) [reply drivers/i2c/i2c_ll_stm32.c]: removed

>

File: drivers/watchdog/wdt_wwdg_stm32.c (hunk 1 above)
Comment (FRASTM) [reply drivers/watchdog/wdt_wwdg_stm32.c]: this is covered by the PullRequest 
driver: watchdog: Incorrect expression in wdt_wwdg_stm32.c [#20617]

**Comment by FRASTM** [comment]
rebase

**Comment by FRASTM** [comment]
rebase

File: drivers/i2c/i2c_ll_stm32.c (hunk 4 above)
Comment (erwango) [reply drivers/i2c/i2c_ll_stm32.c]: Still there as I can see

**Review by erwango — CHANGES_REQUESTED** [review]

File: drivers/i2c/i2c_ll_stm32.c (hunk 4 above)
Comment (FRASTM) [reply drivers/i2c/i2c_ll_stm32.c]: lines 
`#define LOG_LEVEL_CONFIG_I2C_LOG_LEVEL`
`#include <logging/log.h>`
`LOG_MODULE_REGISTER(i2c_ll_stm32);`
do not changed

**Review by erwango — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]

**Pull request merged**